[Semi-Modular] HMS Buffs

SomeRandomOwl opened the pull request:
## About The Pull Request

This is going off a suggestion I have heard from people who use this quirk. Its kinda a good quirk for RP but the side effects of stage 5 as well as the availability of the medicine and unreliability of the medicine isn't so conducive for RP. 

The goal of this PR is simple. Make the consequences of the quirk less harsh, make the stages easier to notice as some people can go from stage 1 - 4 without even getting any symptom, remove the round removal aspect of stage 5 but add in a new effect as well as bringing back old effects stage 5 had that TG removed, and to increase the effectiveness and availability of the medicine for it.

The ways I have achieved the above is as follows:

- I have removed the dusting aspect of the quirk and instead replaced it with a dislocation in the station that crits you
- I have brought back two removed consequences of stage 5 that was previously removed
- I have increased the chances for stage 1-4 symptoms of showing up for the user so they know when they have advanced in a stage
- I have increased the amount of reagents in the pills that you can order and start with.
- I have halved the metabolize rate of the medicine itself
- I have added the pill bottles to the deforest medical vendors that can spawn on the station
- I have created a reagent recipe so chemists can make the medicine itself

## How This Contributes To The Nova Sector Roleplay Experience

The round removal aspect of the quirk isn't very conducive for roleplay, since it implies you are removed from the timeline and people are simply meant to forget you. That's not very role-play centeric. This will give them the option to ask chemists for their medicine, as well as just less harsh ways that it will kill you while also giving interaction with medical in order to revive you.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
![dreamseeker_xt1uo2n2GJ](https://github.com/NovaSector/NovaSector/assets/2568378/0b232c95-3724-4c70-85fa-a440e581c182)

https://github.com/NovaSector/NovaSector/assets/2568378/7af19771-96ee-4157-acd1-12d59b589e29

![dreamseeker_kdjGA7fEpl](https://github.com/NovaSector/NovaSector/assets/2568378/673654fa-c0ee-4912-8bb7-7ea12331716a)

https://github.com/NovaSector/NovaSector/assets/2568378/aef00106-cf50-44a4-b375-fe9e73428077

![dreamseeker_GhD1hNcXwy](https://github.com/NovaSector/NovaSector/assets/2568378/cc018494-e462-4fc7-b078-366a75a473ab)

</details>

## Changelog

:cl:
add: Changed stage 5 of Hereditary manifold Syndrome, it no longer dusts you, but it will still kill careless people
add: Sansufentanyl now has a recipe to make it with chemistry, it is now also added into deforest medical vendors
qol: HMS symptoms are now more noticeable 
qol: Sansufentanyl pills now start with more reagents per pill, as well as halved metabolization rate
/:cl:

Iamgoofball commented:
the removal from the timeline would be funnier if it replaced all attempts to say your character's name with grawlix like in Kingdom Hearts

SpathiCaptain commented:
Longtime HMS quirk user here! I like some of these changes, but feel like others remove a bit of the drama associated with the quirk. Part of the drama of HMS is it's sense of finality for careless mismanagement of stage progression and the reprecussions of doing so without quick rectification. Which can be broken down into a few distinct areas.

- Firstly, the medicine: Sansufentanyl. According to its boilerplate lore, is a proprietary drug manufactured by Interdyne, to act as a fix for the Interdyne created Manifold Sickness. Having more bottles avaliable for easy access is good, but i'm unsure if putting them in the DeForest vendors is the correct choice. Perhaps lowering the price of the experimental medicine crate from cargo, and removing its medical access requirement is better? Furthermore, the ability for the station to create Sansufentanyl (barring unusual methods like strange seeds) absolutely obliterates any theatre of IC issues that a person might have with HMS. (Nor does it really make sense for Interdyne to divulge their formula to NT). If you absolutely _must_ keep the ability for station chemists to create sansu, I suggest something very difficult and time consuming, like: Exotic Stabilizer, Eigenstasium, Liquid Dark Matter, Spaceacillin. Other changes, like making Sansu more effective by halving the metabolize rate are good, but keep in mind that sansu itself has quite a lot of nasty side effects, Sansufentanyl has a lot of really annoying side effects, and making it last four times as long _feels_ like it might be almost worse than the actual HMS symptoms themselves. Perhaps look to tuning down how much taking sansu impacts the player if you are going that route? Or starting HMS patients with an additional bottle of psicodine to help with the extreme side-effect duration.

- Secondly, the dusting aspect. This is what seperates HMS from Brain Tumor, and in my opinion, is totally immutable from the quirk. The hard threat of "I can really die for good!" and the desperation and anxiety from running out or low on Sansufentanyl has created some of the most dramatic, and intense roleplay I have experienced on Nova Sector. In addition, the RP around HMS's permanent and very real consequence of being deleted from the timeline has also created fantastic and dramatic scenarios where my character had to explain the painful result of not having her medicine, and the very real threat that she may simply cease to be entirely if proper care wasn't taken. Removing these aspects neuters the quirk too much, and effectively makes it "Brain Tumor but you sometimes puke blood" and turns Sansufentanyl into "Mannitol but with annoying side effects." Though I personally think the timeline removal is a neat RP aspect of the quirk, and allows for a sort of "If I dust, the character is done" mentality, I can appreciate why people would not like that if taken. Perhaps a config of some sort could be added to select the severity of HMS you would like to have? Or to customize your death message should you actually fail to get medicine before reaching stage five.

- Lastly, Perhaps Interdyne could have some involvement with HMS and Sansufentanyl? I'm unsure exactly how exactly, but both of these (according to whatever the Pandemic says when you analyze HMS anyway) are creations of them. Maybe they could have an extreme surplus of Sansu to sell to the station? Or are able to create it from the chemical dispensers directly to make unique mixes of the drug superior to normal bottles bought from Nanodrug vendors, or cargo? Something to think about, but I feel as if the unique opportunity to add some additional RP layers there might be fun.

SomeRandomOwl commented:
I could look into adding configs for it to adjust the severity for it and alter the symptoms of it likewise if thats wanted if you wish we can bring discussions over here as this is where the suggestion and talk for this took place so you can offer input and talk to other people who use the quirk. I am willing and open to do any suggestion with this so long as its not too too complex @SpathiCaptain 

 https://discord.com/channels/1171566433923239977/1223954303140495523

Seashine commented:
Hello! I made the original suggestion. Overall, I just wanted a separate avenue for acquiring Sansufentanyl as it can be really vexing sometimes - and specifically puts the burden on one department - along with having some pretty unpleasant 'bugs' (a not unreasonable chance for zero symptoms up to a near-lethal state). I am totally willing to flex on how this is solved though!

For example, I actually didn't know IDP made Sansufentanyl! I guess that comes from never having played Virology. There's no (to my knowledge) evidence on this on the IDP side from my experience playing them; they do get a crate of the stuff, but I always rationalized this as a band-aid since they would have to go to the station just to order it, which isn't something they always want to do. If they COULD make Sansufentanyl, that would be really cool, since it would give them more things to sell to the station and more reason to visit. For example, they could round-start make a precursor that NT only gets later, like how they can make advanced precursors to begin with.

Anyways, my intent with this wasn't to ruin the impact of HMS - although having more drastic symptoms before RR would be nice, since it doesn't really have much of an impact until that point. If the progress of the disease could be slowed but the symptoms more severe until finally you get annihilated, that would be awesome, in my mind.

Iamgoofball commented:
honestly i've never gotten the obsession with not round-removing characters when the server has like 50 player slots and respawn enabled, just switch to a different character if you die, y'know? especially if they opted into it with HMS

SomeRandomOwl commented:
Ok so taking in the feedback i received from people who use this quirk Ive gone and done a few things

- I have returned the dusting aspect to it, i just added more to the pool of consequences so its still a threat, but not a guarantee. 
- I have added a goodie bag to cargo that doesnt require medical access, but subsequently is more expensive due to its availability, medical still would be the preferred option as its cheaper and can be department ordered.
- I have removed the ability for the station to make sanusfentinal and instead gave that ability to interdyne, subsequently i added a new reagent which only interdyne gets stored in a roundstart box in their chemistry. 

The ideas behind this is while its more easily avalaible, it also encourages interdyne to make the medicine and try to trade with the station, potentially bargaining and offering cheaper alternatives for the station, While also making their own money on the side from it.

SomeRandomOwl commented:
Obligatory mapping screenshots so @Zergspower or @Jolly-66 doesn't want to rip my head of for not providing mapping screenshots.

<details>
<summary>Screenshots</summary>

![dreamseeker_8mgnrMobFK](https://github.com/NovaSector/NovaSector/assets/2568378/b6a313d7-c9b5-42bf-bf1c-9318b00f7949)

![dreamseeker_JPYrxhuePn](https://github.com/NovaSector/NovaSector/assets/2568378/c07f36e0-6c23-4cb9-85b5-7a705bc69a27)

</details>

SomeRandomOwl commented:
Also additional video of testing the reagent combination.

<details>
<summary>Screenshots</summary>

https://github.com/NovaSector/NovaSector/assets/2568378/7c09bdb8-aed3-4651-9a1f-8840f8156bee

</details>

SomeRandomOwl commented:
Added one bit more, there is a proper secret recipe now that will be stored roundstart in interdynes vault. This contains the recipe for the base reagent this pr adds in the event that interdyne runs out. I feel like this adds a bit more flavor to things with them being the only ones who know the recipe and gives them something to protect.

Obligatory mapping screenshots

<details>
<summary>Screenshots</summary>

![dreamseeker_vpqpi5oIiX](https://github.com/NovaSector/NovaSector/assets/2568378/6e8d42e3-63d0-47b7-bf4c-5e98f89acc1b)

![dreamseeker_Byq5gbx97i](https://github.com/NovaSector/NovaSector/assets/2568378/fcb7768f-34a6-407e-909e-cff5c9a905c4)

</details>

SomeRandomOwl commented:
Those are fighting words bot, i swear

FlufflesTheDog commented on modular_nova/modules/HMS_Changes/sansufentanyl.dm:
```diff
@@ -0,0 +1,106 @@
+	min_catalysts = 1
+	max_catalysts = 1
+	max_input_reagents = 4
+	results = list(/obj/item/reagent_containers/cup/beaker/sansufentanyl_base=20)
```
```suggestion
	results = list(/datum/reagent/medicine/sansufentanyl_base=20)
```

Iamgoofball commented:
>I have removed the ability for the station to make sanusfentinal and instead gave that ability to interdyne, subsequently i added a new reagent which only interdyne gets stored in a roundstart box in their chemistry.

I dunno about this one because it does mean you're SOL once you burn down the station's supply and if nobody is playing on the interdyne today or are busy with extreme roleplay activities on the interdyne.

SomeRandomOwl commented:
> > I have removed the ability for the station to make sanusfentinal and instead gave that ability to interdyne, subsequently i added a new reagent which only interdyne gets stored in a roundstart box in their chemistry.
> 
> I dunno about this one because it does mean you're SOL once you burn down the station's supply and if nobody is playing on the interdyne today or are busy with extreme roleplay activities on the interdyne.

You can still order boxes from cargo? This is just something extra that interdyne can do to interact with the station as they can now actually make the reagent, and a infinite supply of it for free at least for themselves and then they can take it up to the station and sell it themselves for cheeper (or more expensive) than what you can buy from cargo

Iamgoofball commented:
oh based, that works then 👍

SomeRandomOwl commented:
Wait i have one more thing i want to look at for this

SomeRandomOwl commented:
I dont think what i wanted to do is possible to do cleanly so i think its good
